src/psyclone/dynamo0p3.py

```diff
@@ -4140,11 +4163,11 @@ def _upper_bound_fortran(self):
         if self._upper_bound_halo_depth:
             halo_index = str(self._upper_bound_halo_depth)
 
-        # We only require a mesh object if distributed memory is enabled
-        # and the loop is over cells
-        if config.DISTRIBUTED_MEMORY and \
+        # We only require a mesh object to get upper loop bounds if
+        # distributed memory is enabled and the loop is over cells
+        if _CONFIG.distributed_memory and \
            self._upper_bound_name in ["ncells", "cell_halo"]:
-            if self._kern.is_intergrid:
+            if self._kern and self._kern.is_intergrid:
```

Comment:
Could you please explain the reason for adding self._kern here

Reply:
Comment added (built-ins have self._kern == None).